python/fledge/services/core/api/plugins/remove.py

```diff
@@ -0,0 +1,165 @@
+        event = 'Ingest'
+    elif plugin_type == 'filter':
+        event = 'Filter'
+    else:
+        event = 'Egress'
+    payload_data = PayloadBuilder().SELECT('plugin', 'service').WHERE(['event', '=', event]).payload()
+    enabled_asset_list = await storage_client.query_tbl_with_payload('asset_tracker', payload_data)
+    _logger.info(enabled_asset_list['rows'])
+
+    if plugin_type == 'filter':
+        filter_payload = PayloadBuilder().SELECT('name').WHERE(['plugin', '=', plugin_name]).payload()
+        filter_res = await storage_client.query_tbl_with_payload("filters", filter_payload)
+        filter_used = [f['name'] for f in filter_res['rows']]
+        _logger.info(f"filter:{filter_used}")
+    for e in enabled_asset_list['rows']:
+        if (plugin_name == e['plugin'] and plugin_type != 'filter') or (e['plugin'] in filter_used and
+                                                                        plugin_type == 'filter'):
+            get_enabled_plugin = await get_enabled_or_disabled_status(e['service'])
+            _logger.info(get_enabled_plugin)
+            if plugin_name in [x['plugin'] for x in enabled_asset_list['rows']] or e['plugin'] in filter_used:
+                list_of_enabled_plugin.append(e)
+        _logger.info(list_of_enabled_plugin)
+    return list_of_enabled_plugin
+
+
+async def get_enabled_or_disabled_status(service_name: str):
+    """
+        check plugin state(enabled or disabled)
+    """
+    storage_client = connect.get_storage_async()
+    payload_data = PayloadBuilder().SELECT('id', 'enabled').WHERE(['schedule_name', '=', service_name]).payload()
+    enabled_service_list = await storage_client.query_tbl_with_payload('schedules', payload_data)
+    return enabled_service_list['rows']
+
+
+# TODO: notification service state(enabled or disabled)
+async def check_notification_service_state(plugin_type: str, plugin_name: str):
+    pass
+
+
+def purge_plugin(plugin_type: str, name: str):
+    """
+        Remove plugin based on platform
+    """
+    _logger.info("Plugin removal started...")
+    name = name.replace('_', '-').lower()
+    plugin_name = 'fledge-{}-{}'.format(plugin_type, name)
+    stdout_file_path = common.create_log_file(action='delete', plugin_name=plugin_name)
+    get_platform = platform.platform()
+    try:
+        package_manager = 'yum' if 'centos' in get_platform or 'redhat' in get_platform else 'apt'
+        if package_manager == 'yum':
+            cmd = "sudo {} -y remove {} > {} 2>&1".format(package_manager, plugin_name, stdout_file_path)
+        else:
+            cmd = "sudo {} -y purge {} > {} 2>&1".format(package_manager, plugin_name, stdout_file_path)
+        code = os.system(cmd)
+        installed_plugin = PluginDiscovery.get_plugins_installed(plugin_type, False)
+        if name in [plugin['name'] for plugin in installed_plugin] or code != 0:
+            raise KeyError("Plugin is not installed by package manager.".format(name))
```

Comment:
Also handle `PackageError` exception. You may refer custom exception under services/core/api/plugins/exception